src/Core/Content/Seo/SeoUrlGenerator.php

```diff
@@ -63,19 +75,23 @@ public function generate(array $ids, string $template, SeoUrlRouteInterface $rou
             return new RepositoryIterator($repository, $context, $criteria);
         });
 
-        $this->setTwigTemplate($config, $template);
+        $twig = $this->createTwigTemplate($config, $template);
 
-        while ($entities = $iterator->fetch()) {
-            yield from $this->generateUrls($route, $config, $salesChannel, $entities);
+        if ($twig instanceof Environment) {
+            while ($entities = $iterator->fetch()) {
+                yield from $this->generateUrls($twig, $route, $config, $salesChannel, $entities);
+            }
         }
     }
 
-    /**
-     * @internal (flag:FEATURE_NEXT_13410) Parameter $salesChannel will be required
-     */
-    private function generateUrls(SeoUrlRouteInterface $seoUrlRoute, SeoUrlRouteConfig $config, ?SalesChannelEntity $salesChannel, EntityCollection $entities): iterable
-    {
-        $request = $this->requestStack->getMainRequest();
+    private function generateUrls(
+        Environment $twig,
+        SeoUrlRouteInterface $seoUrlRoute,
+        SeoUrlRouteConfig $config,
+        ?SalesChannelEntity $salesChannel,
+        EntityCollection $entities
+    ): iterable {
+        $request = $this->requestStack->getMasterRequest();
```

Comment:
this usage is deprecated

Reply:
I need to lookup where I got that one from. Maybe from my original local fix for shopware 6.3. When this usage is deprecated, what is the new fancy way?

Reply:
Ah I see master vs main. Fixed